### DISCUSSION TITLE
Publishing page modernization one third right section

### DISCUSSION THREAD

**Discussion by kirill-kotovich** [pnp/pnpframework#179] [Q&A]
Hi. 
How can I use one third right  section for transform classic publishing page to modern with custom page layout mapping? No one page layout option except AutoDetect isn't working. If I use in custom page layout mapping xml in <WebParts> section column=1 and column=2 it applies two column layout, but I want to use one third right layout, please help me.

**Comment by jansenbe** [comment]
See https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-model-publishing#pagelayout-element. You need to set the IncludeVerticalColumn attribute and then specify column value 4 to put a transformed web part in the vertical column.

**Reply by kirill-kotovich** [reply]
Thanks, but is not the same that I need. VerticalColumn is one more section, but I need to use one section with layout "one-third right". Is it possible with this conversion tool?

**Reply by jansenbe** [reply]
With the automatic mapping system we'll only switch from the 1/2 + 1/2 to 1/3 + 2/3 model if there's only an image web part in one of the columns. The auto mapping system works the easiest as it allows flexible layouts, if you don't want that you can either choose a fixed (wiki) layout and then you need to use a customized page layout mapping file to ensure all source page elements are assigned to one of columns in the fixed layout. Plan B can be writing your own layoutmanager code and plug that in. See https://docs.microsoft.com/en-us/sharepoint/dev/transform/modernize-userinterface-site-pages-layout for more details.
